### ISSUE TITLE
Changing arguments to position only on an override should be incompatible

### ISSUE THREAD

**Issue by hackaugusto** [python/mypy#13115]
**Bug Report**

**To Reproduce**

`B.f` below should be incompatible with `A.f`:

```python
class A:
    def f(self, a: int) -> None:
        pass


class B(A):
    def f(self, a: int, /) -> None:
        pass
```

because the following fails:

```python
def t(a: A) -> None:
    a.f(a=1)


t(B())
```

**Expected Behavior**

The signature of B.f should cause an error

**Actual Behavior**

No errors are reported

**Your Environment**

- Mypy version used: `mypy 0.961 (compiled: no)` and `mypy 0.980+dev.64a40e8487f591d57576396baaafae9e3a618555 (compiled: no)`
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: Python 3.8.9
- Operating system and version: macOS Monterey 12.4

**Comment by hauntsaninja** [comment]
Agreed that this is unsound, but it's a place where mypy has chosen to be pragmatic. It's not just positional-only args, the same sort of thing also applies to argument name changes:
```
class A:
    def f(self, a: int) -> None: ...

class B(A):
    def f(self, b: int) -> None: ...
```

Linking #1013
